### DISCUSSION TITLE
Statistics Page

### DISCUSSION THREAD

**Discussion by gyCfjSnO** [AdguardTeam/AdGuardHome#3180] [Ideas]
Hi,

I want to suggest a feature/idea.

Can we get a statistics page for the upstream servers and bootstrap servers please?

I would like it to show how many queries each upstream and bootstrap servers have served and what the average response time is for each server.

Kind regards,
A happy AdGuard User

**Comment by gyCfjSnO** [comment]
Bump, I think my idea is great!

**Comment by markhicken** [comment]
You should create a feature request issue if one doesn't already exist. 
https://github.com/AdguardTeam/AdGuardHome/issues/new?assignees=&labels=&template=Feature_request.md

**Comment by Jean-Paul76** [comment]
I added a formal request in #3295 as I too think it would be very useful to gauge the upstream DNS performance. Some stats on Adguard's own cache, such as the ability to check if a specific entry is in the cache, and it's TTL may be useful as well.
